Source/WebCore/WebCoreSwift/CryptoKitShim.swift

```diff
@@ -48,47 +49,123 @@ public final class AesGcm: NSObject {
             var message = UnsafeBufferPointer(start: plainText, count: Int(plainTextSize))
+            let zeroArray = [UInt8](repeating: 0, count: 0)
             if plainTextSize == 0 {
-                zeroArray.withUnsafeBufferPointer { ptr in 
+                zeroArray.withUnsafeBufferPointer { ptr in
                     message = ptr
                 }
             }
-            let symKey = SymmetricKey(data: UnsafeBufferPointer(start: key, count: Int(keySize)))
+            let symKey = SymmetricKey(
+                data: UnsafeBufferPointer(start: key, count: Int(keySize)))
             let sealedBox: AES.GCM.SealedBox
             if additionalDataSize > 0 {
                 sealedBox = try AES.GCM.seal(
                     message,
-                    using: symKey, 
+                    using: symKey,
                     nonce: nonce,
-                    authenticating: UnsafeBufferPointer(start: additionalData, count: Int(additionalDataSize))
+                    authenticating: UnsafeBufferPointer(
+                        start: additionalData, count: Int(additionalDataSize))
                 )
             } else {
                 sealedBox = try AES.GCM.seal(
                     message,
-                    using: symKey, 
+                    using: symKey,
                     nonce: nonce
                 )
             }
-            sealedBox.ciphertext.copyBytes(to: cipherText, count: Int(plainTextSize));
-            sealedBox.tag.copyBytes(to: cipherText + Int(plainTextSize), count: sealedBox.tag.count);
+            sealedBox.ciphertext.copyBytes(to: cipherText, count: Int(plainTextSize))
+            sealedBox.tag.copyBytes(to: cipherText + Int(plainTextSize), count: sealedBox.tag.count)
         } catch {
-            return AesGcmErrorCodes.EncryptionFailed
+            return ErrorCodes.encryptionFailed
         }
-        return AesGcmErrorCodes.Success
+        return ErrorCodes.success
     }
 }
+
+@objc(WebCryptoAesKw)
+public final class AesKw: NSObject {
+    @objc public static func wrap(
+        _ key: UnsafePointer<UInt8>,
+        keySize: UInt,
+        data: UnsafePointer<UInt8>,
+        dataSize: UInt,
+        cipherText: UnsafeMutablePointer<UInt8>,
+        cipherTextSize: UnsafeMutablePointer<UInt64>
+    ) -> ErrorCodes {
+        let cipherTextSizeVal: UInt64 = cipherTextSize.pointee
+        if keySize > Int.max
+            || dataSize > Int.max
+            || cipherTextSizeVal > Int.max
+            || keySize == 0
+            || dataSize == 0
+            || cipherTextSizeVal == 0
+        {
+            return ErrorCodes.invalidArgument
+        }
+        do {
+            let cipher = try AES.KeyWrap.wrap(
+                SymmetricKey(data: UnsafeBufferPointer(start: data, count: Int(dataSize))),
+                using: SymmetricKey(data: UnsafeBufferPointer(start: key, count: Int(keySize))))
+            if cipher.count > Int(cipherTextSizeVal) {
```

Comment:
Does this truncate a uint64_t to an int?  Can we extend instead?

Reply:
It does but I have checked at the top that the value will fit an `Int`. Extending is not an option as the type chosen in Swift for sizes is `Int` and not `UInt` which is causing me to do all this checking.